src/LogFileReader.cpp

```diff
@@ -47,12 +49,44 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    unsigned int allowedCharLength = 15;
+    std::string allowedChars [allowedCharLength] = {"1","2","3","4","5","6","7","8","9","0",":",",",".","-"," "};
+    bool tempBool;
+    for(unsigned int count = 0; count < line.toStdString().length(); count++)
+    {
+        tempBool = false;
+        for(unsigned int count1 = 0; count1 < allowedCharLength; count1++)
+        {
+            if(allowedChars[count1] == line.toStdString().substr(count,1))
+            {
+                tempBool = true;
+            }
+        }
+        if(tempBool == false)
+        {
+            return false;
+        }
+    }
```

Comment:
Instead of doing the error checking yourself, there are built in functions for what you are trying to do. You could split the string by it's delimiters, then for voltage and current take a look at http://doc.qt.io/qt-4.8/qstring.html#toDouble.